tests/test_peer_heap.py

```diff
@@ -107,6 +107,63 @@ def test_update(peer_heap, peers):
     assert peer_heap.peek_peer() == p
 
 
+def test_remove(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = len(peers)
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+
+
+def verify_peer_not_in_heap(peer_heap, p):
+    for peer in peer_heap.peers:
+        assert peer is not p
+
+
+def test_remove_duplicate(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = random.randint(0, len(peers))
+
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        peer_heap.remove_peer(p)
+        peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+        verify_peer_not_in_heap(peer_heap, p)
+
+
+def test_remove_from_empty_heap():
+    heap = PeerHeap()
+    heap.remove_peer(mock_peer())
+
+
+def test_remove_none_exist(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    peer_heap.remove_peer(peers[0])
+
+
+def test_remove_mismatch(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    # create a fake peer with duplicated index.
+    fake_peer = mock_peer()
+    fake_peer.index = 1
+    with pytest.raises(AssertionError):
+        peer_heap.remove_peer(fake_peer)
```

Comment:
No, you need to check that the error you expected was raised. You're not using a unique exception type for this error. You're using a very general exception type. Anything else could raise assertion error too. You need to match on the error message to make sure that it's the assertion error you expected.